ios/CustomerioReactnative.swift

```diff
@@ -118,8 +118,43 @@ class CustomerioReactnative: NSObject {
      */
     private func initializeInApp() -> Void{
         DispatchQueue.main.async {
-            MessagingInApp.shared.initialize()
+            MessagingInApp.shared.initialize(eventListener: self)
         }
     }
 }
 
+extension CustomerioReactnative: InAppEventListener {
+    private func sendEvent(eventType: String, message: InAppMessage, actionValue: String? = nil, actionName: String? = nil) {
+        var body = [
+            "eventType": eventType,
+            "messageId": message.messageId,
+            "deliveryId": message.deliveryId
+        ]
+        if actionValue != nil {
```

Comment:
```suggestion
        if let actionValue = actionValue {
```

This is a common pattern in Swift dev where you check for nil and if not nil, set a locally scoped variable with a non-nil value.